src/java/org/apache/cassandra/db/CounterMutation.java

```diff
@@ -98,7 +99,7 @@ public class CounterMutation implements IMutation
     private static final String LOCK_TIMEOUT_MESSAGE = "Failed to acquire locks for counter mutation on keyspace {} for longer than {} millis, giving up";
     private static final String LOCK_TIMEOUT_TRACE = "Failed to acquire locks for counter mutation for longer than {} millis, giving up";
 
-    private static final Striped<Lock> LOCKS = Striped.lazyWeakLock(DatabaseDescriptor.getConcurrentCounterWriters() * 1024);
+    private static final Striped<Lock> LOCKS = Striped.lock(COUNTER_LOCK_NUM_STRIPES.getInt());
```

Comment:
The previous value was based on the number of available cores. This value doesn't change when we scale the service vertically. Could we use DatabaseDescriptor.getConcurrentCounterWriters() ?